extensions/amp-story/1.0/progress-bar.js

```diff
@@ -164,6 +165,15 @@ export class ProgressBar {
               this.addSegment_(id);
             }
           });
+
+          if (this.segmentCount_ > MAX_SEGMENTS) {
+            this.getInitialFirstExpandedSegmentIndex_(this.activeSegmentIndex_);
+            this.render_(false /** shouldAnimate */);
+          }
+          root.classList.toggle(
+            'i-amphtml-progress-bar-overflow',
+            this.segmentCount_ > MAX_SEGMENTS
+          );
```

Comment:
It looks like this code needs to run every single time a segment is added? Can we write a method vs copy/pasting the snippet?

Reply:
Actually, this will get called initially as well. Removed the other reference to this code.